rcloud.support/R/rcloud.support.R

```diff
@@ -401,117 +433,125 @@ rcloud.search <-function(query, all_sources, sortby, orderby, start, pagesize) {
-        starcount <- response.docs[[i]]$starcount
-        updated.at <- response.docs[[i]]$updated_at
-        user <- response.docs[[i]]$user
-        parts <- response.high[[i]]$content
-        json[i] <- toJSON(c(QTime=time,notebook=notebook,id=id,starcount=starcount,updated_at=updated.at,user=user,numFound=count,pagesize=pagesize,parts=parts,source=as.vector(source)))
-      }
-      return(json)
-    } else
-    return(solr.res$response$docs)
-  } else
-  return(c("error",solr.res$error$msg))
-  }
-
-  if (isTRUE(all_sources)) {
-      main <- query(url)
-      l <- lapply(.session$gist.sources.conf, function(src)
-                  if ("solr.url" %in% names(src)) query(src['solr.url'], src['gist.source']) else character(0))
-      unlist(c(list(main), l))
-  } else query(url)
-}
-
-stitch.search.result <- function(splitted, type,k) {
+                                        response.high[[i]]$content <- gsub("<","&lt;",response.high[[i]]$content)
+                                        response.high[[i]]$content <- gsub(">","&gt;",response.high[[i]]$content)
+                                        response.high[[i]]$content <- gsub("open_b_close","<b style=\\\\\"background:yellow\\\\\">",response.high[[i]]$content)
+                                        response.high[[i]]$content <- gsub("open_/b_close","</b>",response.high[[i]]$content)
+                                        } else
+                                        response.high[[i]]$content <-"[{\"filename\":\"part1.R\",\"content\":[]}]"
+                                      }
+                                      json <- ""
+                                      for(i in 1:length(response.docs)){
+                                        time <- solr.res$responseHeader$QTime
+                                        notebook <- response.docs[[i]]$description
+                                        id <- response.docs[[i]]$id
+                                        starcount <- response.docs[[i]]$starcount
+                                        updated.at <- response.docs[[i]]$updated_at
+                                        user <- response.docs[[i]]$user
+                                        parts <- response.high[[i]]$content
+                                        json[i] <- toJSON(c(QTime=time,notebook=notebook,id=id,starcount=starcount,updated_at=updated.at,user=user,numFound=count,pagesize=pagesize,parts=parts,source=as.vector(source)))
+                                      }
+                                      return(json)
+                                      } else
+                                      return(solr.res$response$docs)
+                                      } else
+                                      return(c("error",solr.res$error$msg))
+                                    }
+
+                                    solr.authKey <- getConf('solr.authkey')
+                                    if(is.null(solr.authKey))solr.authKey=NA 
```

Comment:
For parent SOLR. change authkey to NA since query() does a check for NA's and absent config attributes getConf returns NULLs. Logic was getting twisted for query() to do multiple checks of either NA or NULLs between characters and character vectors (which would return logical(0) instead of false and any() would give TRUE,logical(0) as TRUE. 

In short if its null convert to a NA in a character vector to be consistent with src['solr.authkey']